Kuzzle/API/Controllers/AuthController.cs

```diff
@@ -16,8 +16,9 @@ public sealed class AuthController : BaseController {
     /// Checks the validity of an authentication token.
     /// </summary>
     public async Task<JObject> CheckTokenAsync(string token) {
-      string jwt = api.Jwt;
-      api.Jwt = null;
+      string save = api.AuthenticationToken;
+      api.AuthenticationToken = null;
```

Comment:
As discussed, we don't need the token to be unset here and this can possibly create some failures with some other simultaneous requests.

Reply:
Agreed.

For other reviewers, here is what we discussed with Benoit yesterday: in older SDKs, we took care of removing any previously set authentication token before performing an `auth:checkToken` or an `auth:login` API action.

This means that the SDK and its designers (i.e. us) think that these routes should not be executed while being authenticated.
If that was the case, then we should have designed that feature directly into the API, SDKs should not have an opinion on how Kuzzle's API should behave.

Also, hidden mechanisms such as this one are precisely what we want to kill now, making our products as explicit and straightforward as possible.

Hence: I removed any meddling with the authentication token performed by the Auth controller (apart from memorizing any newly received token from the API).